### ISSUE TITLE
Bearer token not working for DELETE operation and returns error code 2049

### ISSUE THREAD

**Issue by yadro-vavdeev** [nspcc-dev/neofs-node#2000]
User cannot DELETE operations with user BearerToken

```
COMMAND: neofs-cli --config /home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/wallet_config.yml object delete --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8081' --wallet '/home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/TemporaryDir/e485e3e0-0f72-4020-92ad-c4a3cca461cf.json' --cid '9sERC9JvqFWnwNfwQGsSQZm6XDpQRYjRNQoXYsMQ5hmy' --oid '12DdRHARUaPVHyKGKT5TNZTjzEUPvXj1CM1DYTzBXo9J' --bearer '/home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/TemporaryDir/a2a2228a-8d98-4414-aef4-14ced11aa46a'
RETCODE: 2

STDOUT:
failed to get raw object header: read object header via client: status: code = 2048 message = access to object operation denied: access to operation OBJECT_HEAD is denied by extended ACL check: denied by rule
```

All other operations works fine. Sample:
```
COMMAND: neofs-cli --config /home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/wallet_config.yml object hash --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8081' --wallet '/home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/TemporaryDir/e485e3e0-0f72-4020-92ad-c4a3cca461cf.json' --cid '9sERC9JvqFWnwNfwQGsSQZm6XDpQRYjRNQoXYsMQ5hmy' --oid '12DdRHARUaPVHyKGKT5TNZTjzEUPvXj1CM1DYTzBXo9J' --bearer '/home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/TemporaryDir/a2a2228a-8d98-4414-aef4-14ced11aa46a' --range '0:10'
RETCODE: 0

STDOUT:
Offset=0 (Length=10)	: 17a86d8be43d3cdbd8d10858017777a017afb5fca78cfa27653b8bac7eab9dc7
```

HEAD operation with BearerToken 
```
COMMAND: neofs-cli --config /home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/wallet_config.yml object head --rpc-endpoint 'obj-nn-test-node2.spb.yadro.com:8081' --wallet '/home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/TemporaryDir/e485e3e0-0f72-4020-92ad-c4a3cca461cf.json' --cid '9sERC9JvqFWnwNfwQGsSQZm6XDpQRYjRNQoXYsMQ5hmy' --oid '12DdRHARUaPVHyKGKT5TNZTjzEUPvXj1CM1DYTzBXo9J' --bearer '/home/ci/jenkins_home/workspace/Bundle_neofs_build_NB/neofs-testcases/TemporaryDir/a2a2228a-8d98-4414-aef4-14ced11aa46a' --json
RETCODE: 0

STDOUT:
{"objectID":{"value":"AE/b9mx3gwl01hgSp6XQKpLFZuPTHLrSzAvDWVElPkk="}, "signature":{"key":"Avzi1ZI/kzK/s+cE3wF+kAUPXHXYNkCftW/bjGvg/AxS", "signature":"BJb4tAi1iqLGrxJLvRcbydLCAFY6iCNJZIvsR8XdqKLGU6fK9gVrMMCvl67q5ce+8QO5zvW4ZUvSc1AiPw2BtF0=", "scheme":"ECDSA_SHA512"}, "header":{"version":{"major":2, "minor":13}, "containerID":{"value":"g7u05ftfC5VNHmV8uObYkT1/8GXnoCVWs9851RvxKWA="}, "ownerID":{"value":"Ne8yZTDe7fXzOBCyECySa6nNwWXkvt0q3g=="}, "creationEpoch":"6", "payloadLength":"1000", "payloadHash":{"type":"SHA256", "sum":"270LEX7EFCp+HbE/8sVVpkO+Jl9Gs8Ph82X9+MMYMI8="}, "objectType":"REGULAR", "homomorphicHash":{"type":"TZ", "sum":"JzWYZ00ZBjnvOIehcda2X2mGUSO8Fz9p6ox5RB7n6qwgWAHZnX0MUw4hr0dMXFGJYEDF1K4zeetDKItSsppqDg=="}, "sessionToken":{"body":{"id":"s5W/Yls0SCygFY1SzAN6mg==", "ownerID":{"value":"Ne8yZTDe7fXzOBCyECySa6nNwWXkvt0q3g=="}, "lifetime":{"exp":"16", "nbf":"6", "iat":"6"}, "sessionKey":"Avzi1ZI/kzK/s+cE3wF+kAUPXHXYNkCftW/bjGvg/AxS", "object":{"verb":"PUT", "target":{"container":{"value":"g7u05ftfC5VNHmV8uObYkT1/8GXnoCVWs9851RvxKWA="}, "objects":[]}}}, "signature":{"key":"AmGnY5wH01TCnQwiS7x+L6iKhKC4+b5+fMVhJP7Bw/3u", "signature":"BLxOsa/LfNG8IWzH8iB3KCNY2NJuo1WNazSImdv0YlDor0vrwrIg7C49y1J9osiwg0eczaEeBGxXHOPCG06uIJw=", "scheme":"ECDSA_SHA512"}}, "attributes":[{"key":"key1", "value":"val1"}, {"key":"key", "value":"2"}, {"key":"key2", "value":"abc"}, {"key":"FileName", "value":"93908cdc-2aab-447b-8bcb-c522fb591943"}, {"key":"Timestamp", "value":"1667181545"}], "split":null}, "payload":""}
```

TESTS:
https://obj-jenkins.spb.yadro.com/job/neofs-testcases_run/233/allure/#suites/526197b5274bae768c0a631fa726cb06/f68c31bd4bb040b6/

ENV:
```
Build v0.1.1-nb-20221031.1 (31 окт. 2022 г., 4:52:06)
neo-go - 0.99.5-pre-50-g4607b334
neofs-authmate - 0.24.0-32-gf6b94c25
neofs-cli - 0.33.0-91-g98034005
neofs-adm - 0.33.0-91-g98034005
AWS - 2.2.0
```

**Comment by cthulhu-rider** [comment]
:two: :zero: :zero: :zero:

**Comment by fyrchik** [comment]
1. What about not attaching a bearer token to auxiliary requests?
2. Check that HEAD is allowed for a storage node in all cases.
3. Think about opening sessions only with container nodes.
4. Redirect all requests without a session token for non-container nodes?
5. See what we can do with forming better session tokens on the user side. @carpawell 

Validate with 4 cases:
1. session with a container node, delete
2. session with a container node, delete, object is stored locally
3. session with a container node, delete, big object is distributed
4. session with a non-container node, delete

**Comment by carpawell** [comment]
Closed via #2034.
